src/analysis.zig

```diff
@@ -132,6 +140,11 @@ pub const ASTAnalyzer = struct {
             }
         }
 
+        var file_as_struct = @import("rules/file_as_struct.zig").FileAsStruct{};
+
+        // per-tree rules
+        if (self.file_as_struct) try file_as_struct.check_tree(alloc, &faults, file_name, tree);
```

Comment:
hmm, I kind of like the idea of per-tree rules — `check_format` also only runs once per tree, but it's implemented as a per-node rule. This is probably clearer and more efficient; I'll convert check_format to a per-tree rule later.